aspnet/mvc/controllers/filters.rst

```diff
@@ -1,8 +1,327 @@
+
+.. _duration-action-filter:
+
+.. literalinclude:: filters/sample/src/FiltersSample/Filters/DurationActionFilter.cs
+  :language: c#
+  :emphasize-lines: 1,21-26
+  :lines: 8-35
+  :dedent: 4
+
+.. note:: You don't need to use this filter to log action times unless you need to display it on a page. The framework's built-in :doc:`logging </fundamentals/logging>` already logs action timing information.
+
+.. _exception-filters:
+
+Exception Filters
+-----------------
+*Exception Filters* implement either the ``IExceptionFilter`` or ``IAsyncExceptionFilter`` interface.
+
+Exception filters handle unhandled exceptions. They are only called when an exception occurs later in the pipeline. They can provide a single location to implement common error handling policies within an app. The framework provides an abstract `ExceptionFilterAttribute <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/Filters/ExceptionFilterAttribute/index.html>`_ that you should be able to subclass for your needs. Exception filters are good for trapping exceptions that occur within MVC actions, but they're not as flexible as error handling middleware. Prefer middleware for the general case, and use filters only where you need to do error handling *differently* based on which MVC action was chosen.
+
+.. tip:: One example where you might need a different form of error handling for different actions would be in an app that exposes both API endpoints and actions that return views/HTML. The API endpoints could return error information as JSON, while the view-based actions could return an error page as HTML.
+
+Exception filters do not have two events (for before and after) - they only implement ``OnException`` (or ``OnExceptionAsync``). The ``ExceptionContext`` provided in the ``OnException`` parameter includes the ``Exception`` that occurred. If you set ``context.Exception`` to null, the effect is that you've handled the exception, so the request will proceed as if it hadn't occurred (generally returning a 200 OK status). The following filter logs exceptions and marks them as handled:
+
+.. literalinclude:: filters/sample/src/FiltersSample/Filters/LoggingExceptionFilterAttribute.cs
+  :language: c#
+  :emphasize-lines: 7,15,18
+
+.. _result-filters:
+
+Result Filters
+--------------
+*Result Filters* implement either the ``IResultFilter`` or ``IAsyncResultFilter`` interface and their execution surrounds the execution of action results. Result filters are only executed for successful results - when the action or action filters produce an action result. Result filters are not executed when exception filters handle an exception, unless the exception filter sets ``Exception = null``.
+
+.. note:: The kind of result being executed depends on the action in question. An MVC action returning a view would include all razor processing as part of the ``ViewResult`` being executed. An API method might perform some serialization as part of the execution of the result.
```

Comment:
There was some question about razor processing and where that takes place, so I wanted to clarify here where that occurs and also that it only occurs at all for some kinds of actions. Added a link to the action results article, though.